File: backend-spring/src/main/java/io/github/agussyahrilmubarok/backend/service/IAuthService.java
package io.github.agussyahrilmubarok.backend.service;

import io.github.agussyahrilmubarok.backend.model.auth.SignInRequest;
import io.github.agussyahrilmubarok.backend.model.auth.SignInResponse;
import io.github.agussyahrilmubarok.backend.model.auth.SignUpRequest;
import io.github.agussyahrilmubarok.backend.model.auth.SignUpResponse;

public interface IAuthService {
    SignUpResponse signUp(SignUpRequest request);

    SignInResponse signIn(SignInRequest request);
}

File: examples/backend-crud/backend-dotnet/Backend/Application/Service/IAuthService.cs
using Backend.Application.DTO.Auth;
using Backend.Application.DTO.User;

namespace Backend.Application.Service;

public interface IAuthService
{
    Task<AuthResponse> SignUpAsync(SignUpRequest request, CancellationToken ct = default);
    Task<AuthResponse> SignInAsync(SignInRequest request, CancellationToken ct = default);
}